libraries/botbuilder-dialogs/tests/dialogContext.test.js

```diff
@@ -227,203 +243,215 @@ describe('DialogContext', function() {
 
         const dialogs = new DialogSet(dialogState);
-        dialogs.add(new WaterfallDialog('a', [
-            async function (step) {
-                assert(step, `WaterfallStepContext not passed in to WaterfallStep.`);
-                await step.context.sendActivity(`foo`);
-                return Dialog.EndOfTurn;
-            },
-            async function (step) {
-                assert(step, `WaterfallStepContext not passed in to WaterfallStep.`);
-                return await step.endDialog(true);
-            }
-        ]));
-
-        adapter.send(beginMessage)
-            .send(continueMessage)
-            .startTest();
+        dialogs.add(
+            new WaterfallDialog('a', [
+                async function (step) {
+                    assert(step, `WaterfallStepContext not passed in to WaterfallStep.`);
+                    await step.context.sendActivity(`foo`);
+                    return Dialog.EndOfTurn;
+                },
+                async function (step) {
+                    assert(step, `WaterfallStepContext not passed in to WaterfallStep.`);
+                    return await step.endDialog(true);
+                },
+            ])
+        );
+
+        await adapter.send(beginMessage).send(continueMessage).startTest();
     });
 
-    it('should return an error if dialog not found when continue() called.', function (done) {
+    it('should return an error if dialog not found when continue() called.', async function () {
         const adapter = new TestAdapter(async (turnContext) => {
             const dc = await dialogs.createContext(turnContext);
 
             let results;
             try {
                 results = await dc.continueDialog();
-            }
-            catch (err) {
+                if (results.status === DialogTurnStatus.empty) {
+                    await dc.beginDialog('a');
+                }
+                await convoState.saveChanges(turnContext);
+            } catch (err) {
                 assert(err, `Error not found.`);
-                assert.strictEqual(err.message, `DialogContext.continueDialog(): Can't continue dialog. A dialog with an id of 'b' wasn't found.`, `unexpected error message thrown: "${err.message}"`);
+                assert.strictEqual(
+                    err.message,
+                    `DialogContext.continueDialog(): Can't continue dialog. A dialog with an id of 'b' wasn't found.`,
+                    `unexpected error message thrown: "${err.message}"`
+                );
 
                 assert(err instanceof DialogContextError, 'err should be a DialogContextError');
                 assert(err.dialogContext, 'err should include dialogContext');
-
-                return done();
-            }
-            if (results.status === DialogTurnStatus.empty) {
-                await dc.beginDialog('a');
             }
-            await convoState.saveChanges(turnContext);
         });
```

Comment:
I couldn't think of a way to do `assert.rejects()` here since `dc.continueDialog()` is called multiple times. I debated about whether or not I should add a `errorCaught = false => true` boolean to ensure the `catch` block gets hit. Thoughts?

Reply:
[resolved](https://github.com/microsoft/botbuilder-js/pull/3427/commits/8d5fc3a5c6557127a383269a4681c80b69bc66d0)